Format ECK Diagnostics (#6778)
elastic/docs-content: troubleshoot/deployments/cloud-on-k8s/run-eck-diagnostics.md

  - id: cloud-kubernetes
---

> # Capture ECK diagnostics [k8s-take-eck-dump]

Reviewer: ```suggestion
# Run ECK diagnostics tool [k8s-take-eck-dump]
```

Between `run` and `capture` I'd like to get the opinion of other folks of the team (cc: @elastic/admin-docs )

We probably need some alignment and decision here, we have the doc "Capture Elasticsearch diagnostics" with the stack diagnostics tool, but we also have "Run ECE diagnostics tool".

To be honest I'm not a fan of "capture" in this context, but it's just my opinion. For the moment I prefer to suggest the title to be aligned with ECE doc.

If everyone likes more `capture` then I'm ok with it.

Reviewer 2: I think it's fine as is. this area is all a mess so I wouldn't over-index on consistency.

if we did need to change it, we need a `the`

```
# Run the ECK diagnostics tool
```

Author: I believe this is good to leave as "capture" so resolving, but please bump if disagree.

# Capture ECK diagnostics [k8s-take-eck-dump]

> :::{include} /deploy-manage/_snippets/autoops-callout-with-ech.md

Reviewer: @stefnestor : Any reason to remove the `autoops` banner? (cc: @wajihaparvez)

Reviewer 2: ++ please don't remove. product wants these

Author: I can add it back on here & https://github.com/elastic/docs-content/pull/6763 for ECE diags as well, but I'm confused. 

But for my education/confusion. 

Same as the other PR, AutoOps is for Elasticsearch. While ECK _hosts_ Elasticsearch, AutoOps does NOT help with this diagnostic/troubleshooting page at all. It cannot and should NOT replace it at all. So why is it the _first thing_ on this page & why is it on this page specifically at all?

Reviewer 3: This callout was added to a lot of pages as part of a request from product to increase visibility. But it's true, AutoOps isn't directly related to ECK diagnostics. We can remove it from here if it's causing confusion/making users think AutoOps replaces the instructions on this page

Reviewer: I think AutoOps is a more direct replacement & alternative to stack monitoring than ECE/ECK diagnostics.

ECE and ECK diagnostics are primarily and mainly used for platform diagnostics and monitoring, and in a reduced portion for deployments / clusters monitoring.

In this small area (and in very specific use cases) is where AutoOps could be an alternative to an ECE or ECK diagnostic, but just on that, which is not the main purpose of these tools.

So, I agree with @stefnestor , but I couldn't tell PMs that Autoops is not related "at all" with this, as there's a small and indirect relation: ECE and ECK diag tools also take stack diagnostics and we have the autoops banner in the [stack diagnostics doc](https://www.elastic.co/docs/troubleshoot/elasticsearch/diagnostic), so the relation is there.

I'd leave the banners unless approved by PMs. Or we can remove them for the moment and @wajihaparvez could add them back if needed in the future.

Author: As my Support vote: I feel strongly about it not being the first+top thing on the page but don't care if it's on the page. Because that's sending users to the wrong troubleshooting path and they don't even pull what Support asks vs we end up wasting time on random AutoOps setup asks. 

I would be happy to append a "further steps"/recommendations section at the bottom or such on this+ECE diag pages. Or even just leave the blurb as is as a content floating footer instead of header.

Author: > we have the autoops banner in the [stack diagnostics doc](https://www.elastic.co/docs/troubleshoot/elasticsearch/diagnostic), so the relation is there

Also noting FWIW: @eedugon, you linked the ES Diagnostic page. And AutoOps IS expected to replace it long term 😄, so 100% I expect it to show on that page. As counter "stack diagnostic" example, the [KB Diagnostic](https://www.elastic.co/docs/troubleshoot/kibana/capturing-diagnostics) does not mention AutoOps. And AutoOps current form would be pointless to monitor KB and useless to replace its diagnostic. Same for [LS Diagnostic](https://www.elastic.co/docs/troubleshoot/ingest/logstash/diagnostic).

Author: If it works for y'all, I've shifted it down the page further to colocate near `--stack-diagnostics-timeout` which is about ES+KB stack diagnostics. Please LMK if you want it to remain top-of-page.

Reviewer: @stefnestor , I completely agree with you, and I also think that on this type of page, the Autoops banner should probably be a bit different than the rest of the banners, giving a bit more context and without confusion.

Technically speaking, Autoops could help users to do a better monitoring of their systems and avoiding the need to troubleshoot with the diagnostics tool if the issues are prevented :-D

I'll review the PR now, but I'm also ok with removing the banner as @wajihaparvez said it was ok with her too.

## Prepare [k8s_prepare] 

To install the [eck-diagnostics](https://github.com/elastic/eck-diagnostics/) tool:

1. Locate the [latest release](https://github.com/elastic/eck-diagnostics/releases/latest).
2. Download the `*.tar.gz` asset associated to your hardware profile.
3. Unpack the gzip'ed tar archive to access the diagnostic binary `eck-diagnostics`.

## Run [k8s_run] 

> You must know the Kubenetes namespaces of your operator and stack resources to run this diagnostic. The most common usage of this `eck-diagnostics` binary is:

Author: ```suggestion
You must know the Kubernetes namespaces of your operator and stack resources to run this diagnostic. The most common usage of this `eck-diagnostics` binary is:
```

```bash
eck-diagnostics -o <operator-namespaces> -r <resources-namespaces>
```

This tool supports various command line flags. The most common command line flags considered are:

* (Optional) Run it with `-h` or `--help` to print all available options. 

    ```bash
    eck-diagnostics --help
    ```

> * (Required) Run it with `-r` or `--resources-namespace` to indicate the namespaces where your Elastic stack resources are deployed. 

Author: ```suggestion
* (Required) Run it with `-r` or `--resources-namespaces` to indicate the namespaces where your Elastic stack resources are deployed. 
```

* (Optional) Run it with `-o` or `--operator-namespaces` to override the default `elastic-system` namespace for the deployed ECK operator.

* (Optional) Run it with `--stack-diagnostics-timeout` to designate the maximum waiting time to pull the {{es}} and {{kib}} diagnostics. Defaults to `5m0s`.

> * (Optional) Run it with `--run-stack-diagnostics=false` to disable automatically pulling [{{es}} diagnostics](/troubleshoot/elasticsearch/diagnostic.md) and [{{kib}} diagnostics](/troubleshoot/kibana/capturing-diagnostics.md) hosted within the resources namespace. This is enabled by default and is recommended. However, this requires the temporary deployment of additional Pods into the Kubernetes cluster. 

Author: ```suggestion
* (Optional) Run it with `--run-stack-diagnostics=false` to deactivate pulling [{{es}} diagnostics](/troubleshoot/elasticsearch/diagnostic.md) and [{{kib}} diagnostics](/troubleshoot/kibana/capturing-diagnostics.md) hosted within the resources namespace. This is enabled by default and is recommended. However, this requires the temporary deployment of additional Pods into the Kubernetes cluster. 
```

> * (Optional) Run it with `--run-stack-diagnostics=false` to deactivate collecting [{{es}} diagnostics](/troubleshoot/elasticsearch/diagnostic.md) and [{{kib}} diagnostics](/troubleshoot/kibana/capturing-diagnostics.md) from the resources namespace. This is enabled by default and is recommended, but requires the temporary deployment of additional Pods into the Kubernetes cluster. 

Reviewer: I'm ok with the new format in general, but I'd suggest to not start every bullet with `Run it with`.

If we introduce the list of bullets with "The most command line flags are:" then the bullets should directly present the flags and be in alignment with the introductory sentence.

> * (Optional) Check [ECK Diagnostics in air-gapped environments](/deploy-manage/deploy/cloud-on-k8s/air-gapped-install.md#k8s-eck-diag-air-gapped) for command line flags to run support diagnostics in environments without access to the open internet.

Reviewer: If that item doesn't really present a command line flag it should probably be presented at the end of the list without being a bullet. Or starting the bullet with For extra command line flags....

> * (Optional) The tool can also filter the Elastic resources that it runs diagnostics against by specifying the `-f` or `--filters` flag. By specifying the type and name of resource, you can filter for any combination of Elastic stack components.

Reviewer: I'd update the format of the bullet and align it with the rest of the bullets, so the bulleted list feels better structured.

## Prepare [k8s_prepare] 

To install the [eck-diagnostics](https://github.com/elastic/eck-diagnostics/) tool:

1. Locate the [latest release](https://github.com/elastic/eck-diagnostics/releases/latest).
2. Download the `*.tar.gz` asset associated with your hardware profile.
3. Unpack the gzip'ed tar archive to access the diagnostic binary `eck-diagnostics`.

## Run [k8s_run] 

You must know the Kubernetes namespaces of your operator and stack resources to run this diagnostic. The most common usage of this `eck-diagnostics` binary is:

```bash
eck-diagnostics -o <operator-namespaces> -r <resources-namespaces>
```

This tool supports various command line flags. The most common command line flags are:

* (Optional) `-h` or `--help` to print all available options. 

    ```bash
    eck-diagnostics --help
    ```

* (Required) `-r` or `--resources-namespaces` to indicate the namespaces where your Elastic stack resources are deployed. 

> * (Optional) `-o` or `--operator-namespaces` to override the default `elastic-system` namespace for the deployed ECK operator.

Reviewer: ```suggestion
* (Optional) `-o` or `--operator-namespaces` to specify the namespace where the ECK operator is installed. Defaults to `elastic-system`.
```

* (Optional) `--stack-diagnostics-timeout` to designate the maximum waiting time to pull the {{es}} and {{kib}} diagnostics. Defaults to `5m0s`.

* (Optional) `--run-stack-diagnostics=false` to deactivate collecting [{{es}} diagnostics](/troubleshoot/elasticsearch/diagnostic.md) and [{{kib}} diagnostics](/troubleshoot/kibana/capturing-diagnostics.md) from the resources namespace. This is enabled by default and is recommended, but requires the temporary deployment of additional Pods into the Kubernetes cluster.

> * (Optional) `-f` or `--filters` to filter the Elastic resources that it runs diagnostics against. By specifying the type and name of resource, you can filter for any combination of Elastic stack components.

Reviewer: ```suggestion
* (Optional) `-f` or `--filters` to filter the Elastic stack resources to run diagnostics against. By specifying the type and name of resource, you can filter for any combination of Elastic stack components.
```
